pyhap/accessory.py

```diff
@@ -273,6 +290,43 @@ def remove_paired_client(self, client_uuid):
     def paired(self):
         return len(self.paired_clients) > 0
 
+    @property
+    def xhm_uri(self):
+        """Generates the X-HM:// uri (Setup Code URI)
+        @rtype: str
+        """
+        buffer = bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00')
+
+        value_low = int(self.pincode.replace(b'-', b''), 10)
+        value_low |= 1 << 28
+        struct.pack_into('>L', buffer, 4, value_low)
+
+        if self.category & 1:
+            buffer[4] = buffer[4] | 1 << 7
+
+        value_high = self.category >> 1
+        struct.pack_into('>L', buffer, 0, value_high)
+
+        encoded_payload = base36.dumps(struct.unpack_from('>L', buffer, 4)[0]
+                                       + (struct.unpack_from('>L', buffer, 0)[0] * (2**32))).upper()
+        while len(encoded_payload) < 9:
+            encoded_payload = '0' + encoded_payload
+
+        return 'X-HM://' + encoded_payload + self.setup_id
+
+    @property
+    def qr_code(self):
+        """Generates a pyqrcode object which can be used to store SVG, or print to
```

Comment:
How about: Generate a QR code for pairing with this accessory.

Reply:
No problem.